app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -0,0 +1,555 @@
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentSender;
+import android.content.ServiceConnection;
+import android.content.pm.PackageManager;
+import android.net.ConnectivityManager;
+import android.net.wifi.WifiConfiguration;
+import android.os.Build;
+import android.os.IBinder;
+import android.util.Log;
+import android.widget.Button;
+import android.widget.TextView;
+import android.widget.Toast;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
+import android.os.Bundle;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+import androidx.recyclerview.widget.RecyclerView;
+import butterknife.BindView;
+import com.google.android.gms.common.api.ApiException;
+import com.google.android.gms.common.api.ResolvableApiException;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.location.LocationSettingsRequest;
+import com.google.android.gms.location.LocationSettingsResponse;
+import com.google.android.gms.location.LocationSettingsStates;
+import com.google.android.gms.location.LocationSettingsStatusCodes;
+import com.google.android.gms.tasks.Task;
+import io.reactivex.Flowable;
+import io.reactivex.SingleObserver;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.Disposable;
+import java.io.File;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import javax.inject.Inject;
+import kotlin.Unit;
+import kotlin.jvm.functions.Function0;
+import org.kiwix.kiwixmobile.R;
+import org.kiwix.kiwixmobile.base.BaseActivity;
+import org.kiwix.kiwixmobile.utils.AlertDialogShower;
+import org.kiwix.kiwixmobile.utils.KiwixDialog;
+import org.kiwix.kiwixmobile.utils.ServerUtils;
+import org.kiwix.kiwixmobile.wifi_hotspot.HotspotService;
+import org.kiwix.kiwixmobile.zim_manager.fileselect_view.SelectionMode;
+import org.kiwix.kiwixmobile.zim_manager.fileselect_view.adapter.BookOnDiskDelegate;
+import org.kiwix.kiwixmobile.zim_manager.fileselect_view.adapter.BooksOnDiskAdapter;
+import org.kiwix.kiwixmobile.zim_manager.fileselect_view.adapter.BooksOnDiskListItem;
+
+import static org.kiwix.kiwixmobile.utils.StyleUtils.dialogStyle;
+
+public class ZimHostActivity extends BaseActivity implements
+    ZimHostCallbacks, ZimHostContract.View {
+
+  @BindView(R.id.startServerButton)
+  Button startServerButton;
+  @BindView(R.id.server_textView)
+  TextView serverTextView;
+  @BindView(R.id.recycler_view_zim_host)
+  RecyclerView recyclerViewZimHost;
+
+  @Inject
+  ZimHostContract.Presenter presenter;
+
+  @Inject AlertDialogShower alertDialogShower;
+
+  public static final String ACTION_TURN_ON_AFTER_O = "Turn_on_hotspot_after_oreo";
```

Comment:
These actions probabky all belong in the Service

Reply:
Fixed. 0e75039689c817e1eb2359fc43ed290106fd0807